tuf/data/root.go

```diff
@@ -43,10 +43,11 @@ func NewRoot(keys map[string]PublicKey, roles map[string]*RootRole, consistent b
 
 // ToSigned partially serializes a SignedRoot for further signing
 func (r SignedRoot) ToSigned() (*Signed, error) {
-	s, err := json.MarshalCanonical(r.Signed)
+	s, err := defaultSerializer.MarshalCanonical(r.Signed)
```

Comment:
In general I like the idea of isolating the JSON marshaling/unmarshaling into its own package, it's nice to keep that logic separate from the TUF objects themselves